.gitignore

```diff
@@ -39,4 +39,8 @@ next-env.d.ts
 # Ignore generated credentials from google-github-actions/auth
 gha-creds-*.json
 
+# IDE
+.idea
+.vscode
```

Comment:
What's the rationale for gitignoring .vscode?

Reply:
Some IDEs will autogenerate a .<ide> file with workspace specific preferences, like code style configuration, debug configuration, vcs settings, plugin configuration, and other cached behavior to apply for the workspace. These are specific to the developer's IDE, workspace and preferences, and shouldn't be enforced on other local machines.

Best practice to just .gitingore them so as to not pollute the repo.

Reply:
Yeah, I can see that, but I figured the dev would have put their preferred configs on a global level rather than workspace. 

You can gitignore .ide and other things, but perhaps leave .vscode out of it. I think it's handy to have the extensions.json and settings.json as they serve as extension recommendations (which the dev can decline should they like) and some pretty essential markdown, mdx, plaintext configs (because otherwise the mdx files become impossible to work with given our literary purposes with them). I think for vscode the dev can overrride workspace settings with their own user settings? If it turns out to be an issue down the line, we can gitignore it.